src/luaharfbuzz/luaharfbuzz.c

```diff
@@ -10,21 +10,20 @@ int shape_full (lua_State *L) {
   unsigned int num_features = lua_rawlen(L, 3);
   Feature *features = (Feature *) malloc (num_features * sizeof(hb_feature_t));
 
-  lua_pushnil(L);
-  while (lua_next(L, 3) != 0) {
+  for (i = 0; i != num_features; ++i) {
```

Comment:
I’d declare `i` here to limit its scope.

Reply:
```suggestion
  for (local i = 0; i != num_features; ++i) {
```